.github/workflows/checks.yml

```diff
@@ -0,0 +1,97 @@
+name: Quality checks and tests
+on:
+  push:
+  pull_request:
+    branches: [ main, master ]
```

Comment:
Note: some of these are intentionally left as-is for testing purposes and will be removed before merging.